Feat app 670 integrate new user personal dahsboard

alexander-astrand opened the pull request:
## Description

https://regennetwork.atlassian.net/browse/APP-670
https://regennetwork.atlassian.net/browse/APP-697

<!-- Add a description of the changes that this PR introduces and the files that
are the most critical to review. -->

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [x] provided a link to the relevant issue or specification
- [x] provided instructions on how to test
- [x] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### How to test

<!-- This should include steps on how to test the fixes or features within the marketplace app and/or the website.
If applicable, this should also include links to the created/updated components on storybook. -->

1. Make sure all the different sections lead to the correct page in the new dashboard.
2. Check that designs and mobile ui is correct
3. Check that all the routes now lead to /dashboard for the new dashboard and /profiles for new profile page

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

erikalogie commented:
Great work on this!

Here is some feedback:
- On the projects page in the dashboard, clicking on "view on personal profile" should take the user to the projects tab of the personal profile.
- The issued credit batches page in the dashboard should have a button that links to "view on personal profile"
- The tables in the bridge tab do not seem to be getting data, as I thought this shared testing account did have both bridgable and bridged credits:
<img width="1682" alt="Monosnap Regen Marketplace : Buy, trade and retire carbon credits | Regen Marketplace : Buy, trade and retire carbon credits 2025-06-23 09-54-58" src="https://github.com/user-attachments/assets/bb81a8e3-5a1b-4eda-8969-bec9a90b6b23" />

- The navigation along the side should be selected in accordance with the current page the user is viewing:
<img width="1708" alt="Monosnap Regen Marketplace : Buy, trade and retire carbon credits | Regen Marketplace : Buy, trade and retire carbon credits 2025-06-23 09-49-21" src="https://github.com/user-attachments/assets/63eb369f-f68e-4553-b04a-60cfc220f884" />
- This table has some rounded corners in the wrong place (same on credit batches table):
<img width="1404" alt="Monosnap Regen Marketplace : Buy, trade and retire carbon credits | Regen Marketplace : Buy, trade and retire carbon credits 2025-06-23 09-48-45" src="https://github.com/user-attachments/assets/570a1182-bad0-4a84-b645-0ffe932b5ddf" />
- When I click on one of the "view on personal profile" buttons, I can't click the back button to go back to the dashboard easily. It does work the other way around (i.e. if I click "manage portfolio" from the profile and go to the dashboard, then click the back button, I am back at the user profile).

blushi commented:
> @erikalogie All the notes you had should be fixed now.
> 
> When it comes to the bridgeable credits, i don't think they have been visible with the testing account during the time that I have worked with the profile and dashboard. I only see the bridged ecocredits, and on dev there is neither bridged nor bridgeable at the moment.

You see some bridged ecocredits because you are still using redwood testnet for local dev while on dev/deploy previews, we now use the upgrade testnet so no bridged ecocredits for the testing account on this new testnet
Also bridging on staging has not been working for some time now since we haven't updated our bridge staging service to use the new polygon testnet and deploy needed contracts there nor the new upgrade testnet. I guess it's alright since the plan is to shut it down.

erikalogie commented:
Looking at it on my mobile phone, I think that the mobile navigation has too small a font size (this is my bad). Instead of 12px, can we go with 16px? As it is here: https://www.figma.com/design/L3VpDxt1PXTATPbSY0Z7DV/Organizations--org-member-permissions--and-project-collaborator-permissions?node-id=3968-182710&t=dWkES8oGU8s1pTuL-1. Same applies to the name of the user within the dropdown. 

Similarly, can we make the person's name on the main navigation on mobile 16px instead of 12px? It feels very small right now.

I'm getting this error when I click on the portfolio tab on mobile while logged in with email only:
![1000206484](https://github.com/user-attachments/assets/71eb83ba-6104-4ca9-b509-b5dec27304e4). The user can still purchase ecocredits even if they do not have a Keplr wallet, so the portfolio tab should be accessible.

When clicking on "personal dashboard" on mobile from the user profile page, in this account where I am logged in only with email, I end up on the "projects" page by default instead of the portfolio page.

When logged in only with email, the email address should show in the space where the "..." is currently, and the "copy to clipboard" icon should disappear
![1000205699](https://github.com/user-attachments/assets/d8c5f991-ad6d-4095-9f9d-58a76bf98ad9)

I'm also 
![1000206486](https://github.com/user-attachments/assets/8c0bb1e5-bff3-4fdf-90a7-a835a0ecee9d)
noticing that the background does not extend the full height of the mobile device, as in here:


Can we also fix this when there is an overlap that a light drop shadow appears? 
<img width="587" alt="Monosnap Regen Marketplace 2025-06-26 07-59-56" src="https://github.com/user-attachments/assets/717661be-40aa-4a12-bc3e-ab7a75ac3c32" />

Can we make each of the buttons less tall? I feel like there is too much space.
<img width="577" alt="Monosnap Regen Marketplace 2025-06-26 08-08-22" src="https://github.com/user-attachments/assets/46bcfdb4-c492-485d-9aac-103d2354453e" />
<img width="1695" alt="Monosnap Regen Marketplace 2025-06-26 08-11-33" src="https://github.com/user-attachments/assets/06a87277-2423-493c-9e56-2fc971695295" />


> @erikalogie All the notes you had should be fixed now.
> 
> When it comes to the bridgeable credits, i don't think they have been visible with the testing account during the time that I have worked with the profile and dashboard. I only see the bridged ecocredits, and on dev there is neither bridged nor bridgeable at the moment.

The tabs for ecocredits and bridge should disappear altogether in this case, as it is here: https://www.figma.com/design/L3VpDxt1PXTATPbSY0Z7DV/Organizations--org-member-permissions--and-project-collaborator-permissions?node-id=2780-1230281&t=I8NcwMFZLjAPCy49-1

erikalogie commented on web-marketplace/src/pages/Dashboard/Dashboard.BannerCard.tsx:
```diff
@@ -1,52 +0,0 @@
-    <>
-      {content && !profileBannerCard[activeAccount?.id] ? (
-        <div className="pb-[36px] sm:pb-20">
-          <BannerCard
```
let's get rid of it

alexander-astrand replied:
@blushi  I think it makes more sense to create a separate task for this!

alexander-astrand commented:
@erikalogie The portfolio tab when connected by email might also need to be a separate task. The current component, that is basically the same as the old dashboard, is heavily reliant on a wallet. I'm also not sure what is available to manage when you have only an email address?

erikalogie commented:
> > Looking at it on my mobile phone, I think that the mobile navigation has too small a font size (this is my bad). Instead of 12px, can we go with 16px? As it is here: https://www.figma.com/design/L3VpDxt1PXTATPbSY0Z7DV/Organizations--org-member-permissions--and-project-collaborator-permissions?node-id=3968-182710&t=dWkES8oGU8s1pTuL-1. Same applies to the name of the user within the dropdown.
> 
> Would you like to have 16px on both mobile and desktop? Currently it's 14px on desktop

I think the 14px on desktop looks fine to my eyes, its just the mobile that looks tiny

blushi commented on web-marketplace/src/components/organisms/DashboardNavigation/DashboardNavigation.MobileHeader.tsx:
```diff
@@ -0,0 +1,64 @@
+        'md:hidden fixed top-0 w-[100%] z-40 flex gap-10 bg-bc-neutral-0 px-5 py-10 transition-all duration-300',
+        mobileMenuOpen && 'blur-sm',
+      )}
+      style={{ borderBottom: '1px solid #e5e7eb' }}
```
let's use tailwind classes instead and avoir using raw color hex as this won't change with the theme

blushi replied:
you need to specific `border border-solid border-{color}`

blushi commented on web-marketplace/src/pages/Dashboard/Dashboard.tsx:
```diff
@@ -73,107 +78,270 @@ const Dashboard = (): JSX.Element => {
+                    {/* Mobile-only subtitle */}
+                    <div className="block md:hidden mb-2">
+                      <span
+                        style={{
+                          fontFamily: 'Mulish',
+                          fontWeight: 800,
+                          fontSize: '10px',
+                          lineHeight: '100%',
+                          letterSpacing: '1px',
+                          textTransform: 'uppercase',
+                        }}
```
ditto let's convert this to tailwind css or just use TextButton component

blushi reviewed (commented):
- when loading the dashboard portfolio, I see the portfolio/bridge tabs for a short amount of time before it disappears
<img width="448" alt="image" src="https://github.com/user-attachments/assets/d51557b5-8088-4621-94ba-2f1de81155b0" />

- it might be nice to have some smoother animation when we open/close the left nav, wdty @erikalogie

blushi commented on web-marketplace/src/components/organisms/DashboardNavigation/DashboardNavigation.Header.tsx:
```diff
@@ -25,20 +29,25 @@ export const DashboardNavHeader = ({
+      ...activeAccount,
+      type:
+        activeAccount.type === 'user'
+          ? (window as any).AccountType?.User || 'USER'
```
fixed here, but still the same in `DashboardNavigationMobileHeader`

alexander-astrand replied:
This should be sorted, I refactored that right after the first comment!

alexander-astrand replied:
Ahh I missed that it was in two places, should be fixed now!

erikalogie commented:
> * when loading the dashboard portfolio, I see the portfolio/bridge tabs for a short amount of time before it disappears
> 
> <img alt="image" width="448" src="https://private-user-images.githubusercontent.com/3149358/461062874-d51557b5-8088-4621-94ba-2f1de81155b0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTEzNzg4NjMsIm5iZiI6MTc1MTM3ODU2MywicGF0aCI6Ii8zMTQ5MzU4LzQ2MTA2Mjg3NC1kNTE1NTdiNS04MDg4LTQ2MjEtOTRiYS0yZjFkZTgxMTU1YjAucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDcwMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA3MDFUMTQwMjQzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NzM3NDA0NjgxNDkzMDRkZGZlNDg3Zjc0N2Q3ZTM4NDVhYjA3NmQwNTU1MTQ4MzUyNGVhZDcyZTc5ZDllYzI2ZCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.9wA447SbW5HKJ_ZJ7nt5l7bcjhTFRmYdE9TMIluI_J8">
> * it might be nice to have some smoother animation when we open/close the left nav, wdty @erikalogie

Yes agreed about the smoother animation

erikalogie commented:
After I make some change on the "edit personal profile" page and then click "save", it redirects me to the "portfolio" page. I believe this used to redirect to the user profile page automatically, though @r41ph made a comment that he thought this was a weird interaction. Maybe we should just have it save and stay on the current edit form?

erikalogie commented:
<img width="849" alt="Monosnap Regen Marketplace 2025-07-01 11-05-52" src="https://github.com/user-attachments/assets/6f6f6a95-5ec1-4750-a1d7-7c0ccc3cc0d4" />

Small design thing on the sell orders page

erikalogie commented:
<img width="354" alt="Monosnap Regen Marketplace 2025-07-01 11-12-17" src="https://github.com/user-attachments/assets/69ff32a4-9891-4147-954d-ff07955aa332" />

alexander-astrand commented on web-marketplace/src/pages/Dashboard/Dashboard.tsx:
```diff
@@ -73,107 +68,250 @@ const Dashboard = (): JSX.Element => {
+                            ...activeAccount,
+                            addr: activeAccount.addr ?? undefined,
+                          }
+                        : undefined
```
Hmm I think I refactored the ViewProfile to handle this better but must have forgot to remove this. Fixed it now!

alexander-astrand commented:
> I tested this while connected with Wallet Connect (I added the deploy preview URL to the list of allowed URLs on cloud.reown.com) and when accessing /dashboard, I just get a blank page

Yeah, same for me! Will have to look into this once I get WalletConnect working in my local environment!

erikalogie commented:
Looks like the deploy preview link is not working currently:
<img width="988" alt="Monosnap feat: APP-670 integrate new user personal dashboard by alexander-astrand · Pull Request #2674 · regen-network:regen-web 2025-07-07 13-21-18" src="https://github.com/user-attachments/assets/c3f227c7-a594-4c9c-a524-573e2d7ccf6d" />

erikalogie commented:
I'm wondering if this animation can be smoother? It looks a little weird how the text comes in, it would be nice if the text didn't move around so much. https://www.loom.com/share/6ec487d522304ae994de993d14bf7e7d

S4mmyb commented:
> > I'm wondering if this animation can be smoother? It looks a little weird how the text comes in, it would be nice if the text didn't move around so much. https://www.loom.com/share/6ec487d522304ae994de993d14bf7e7d
> 
> Yeah I was trying to make it look better but couldn't find a good solution, I sped up the transition a little bit which I thought looked better! But I can take a look again and see if I can delay the text appearing when expanding!

It also feels like there's a delay from 'on-click' to this actually collapsing.

blushi reviewed (commented):
looks great, a few nits still

blushi reviewed (approved):
LGTM

alexander-astrand commented:
> > https://regennetwork.atlassian.net/browse/APP-737 Should be covered here now as well!
> 
> @alexander-astrand Erika just provided the design for it: https://www.figma.com/design/L3VpDxt1PXTATPbSY0Z7DV/Organizations--org-member-permissions--and-project-collaborator-permissions?node-id=9783-119267&t=7Edkdl9jixJW4zFJ-0, it's for the public profile page so I guess you got confused

I already implemented something similar, it's only the background color and text that needs to be changed!

blushi commented:
> > > https://regennetwork.atlassian.net/browse/APP-737 Should be covered here now as well!
> > 
> > 
> > @alexander-astrand Erika just provided the design for it: https://www.figma.com/design/L3VpDxt1PXTATPbSY0Z7DV/Organizations--org-member-permissions--and-project-collaborator-permissions?node-id=9783-119267&t=7Edkdl9jixJW4zFJ-0, it's for the public profile page so I guess you got confused
> 
> I already implemented something similar, it's only the background color and text that needs to be changed! I'll fix it now!

Ok this is one account profile without projects, we miss the projects tab but I guess this can be addressed later since this PR is already quite large.
But could you run i18n scripts? looks like there are some missing translations:

![image](https://github.com/user-attachments/assets/63cf333f-c09f-45f2-90f9-ddeec1a31394)

alexander-astrand commented:
> > > > https://regennetwork.atlassian.net/browse/APP-737 Should be covered here now as well!
> > > 
> > > 
> > > @alexander-astrand Erika just provided the design for it: https://www.figma.com/design/L3VpDxt1PXTATPbSY0Z7DV/Organizations--org-member-permissions--and-project-collaborator-permissions?node-id=9783-119267&t=7Edkdl9jixJW4zFJ-0, it's for the public profile page so I guess you got confused
> > 
> > 
> > I already implemented something similar, it's only the background color and text that needs to be changed! I'll fix it now!
> 
> Ok this is one account profile without projects: https://deploy-preview-2674--regen-marketplace.netlify.app/profiles/regen1gtlfmszmhv3jnlqx6smt9n6rcwsfydrhznqyk9/projects, we miss the projects tab but I guess this can be addressed later since this PR is already quite large. But could you run i18n scripts? looks like there are some missing translations:
> 
> ![image](https://private-user-images.githubusercontent.com/3149358/463747968-63cf333f-c09f-45f2-90f9-ddeec1a31394.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTE5OTE4NDAsIm5iZiI6MTc1MTk5MTU0MCwicGF0aCI6Ii8zMTQ5MzU4LzQ2Mzc0Nzk2OC02M2NmMzMzZi1jMDlmLTQ1ZjItOTBmOS1kZGVlYzFhMzEzOTQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDcwOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA3MDhUMTYxOTAwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9Yjg1MjVlZGU2M2I1MWNjMGFiZmI0ZTZmNmNmNmE1YWZhNTVhNjc3MjQ3ZjRkODEyODM5ZjNkMzlkMzcxNTMyMSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.hBjcvh-5GPsUY2CgrPxqCIWe5yTVy52wbXiZ0v4w2L8)

I pushed the translations and updated the empty state for email! I agree if we want to do more modifications to that issue it's better to save it for another PR! 

Will you take another look @erikalogie before merging?

erikalogie reviewed (approved):
LGTM
